packages/inter-protocol/src/proposals/replaceElectorate.js

```diff
@@ -0,0 +1,332 @@
+    installation: {
+      consume: { committee },
+    },
+    instance: {
+      produce: { economicCommittee },
+    },
+  },
+  { options: { committeeName, committeeSize } },
+) => {
+  const COMMITTEES_ROOT = 'committees';
+
+  trace('startNewEconomicCommittee');
+
+  trace(`committeeName ${committeeName}`);
+  trace(`committeeSize ${committeeSize}`);
+
+  const committeesNode = await makeStorageNodeChild(
+    chainStorage,
+    COMMITTEES_ROOT,
+  );
+  const storageNode = await E(committeesNode).makeChildNode(
+    sanitizePathSegment(committeeName),
+  );
+
+  const marshaller = await E(board).getPublishingMarshaller();
+
+  trace('Starting new EC Committee Instance');
+
+  const privateArgs = {
+    storageNode,
+    marshaller,
+  };
+
+  const terms = {
+    committeeName,
+    committeeSize,
+  };
+
+  const startResult = await E(startUpgradable)({
+    label: 'economicCommittee',
+    installation: committee,
+    privateArgs,
+    terms,
+  });
+
+  const { instance, creatorFacet } = startResult;
+
+  trace('Started new EC Committee Instance Successfully');
+
+  economicCommitteeKit.reset();
+  economicCommitteeKit.resolve(
+    harden({ ...startResult, label: 'economicCommittee' }),
+  );
+
+  economicCommittee.reset();
+  economicCommittee.resolve(instance);
+
+  economicCommitteeCreatorFacet.reset();
+  economicCommitteeCreatorFacet.resolve(creatorFacet);
+
+  return creatorFacet;
+};
+
+/**
+ * Replaces the electorate for governance contracts by creating a new Economic
+ * Committee and updating contracts with the new electorate's creator facet.
+ *
+ * @param {EconomyBootstrapPowers} permittedPowers - The resources and
+ *   capabilities needed for operations, including access to governance
+ *   contracts and the PSM kit.
+ * @param {{
+ *   options: {
+ *     committeeName: string;
+ *     voterAddresses: Record<string, string>;
+ *     highPrioritySendersConfig: {
+ *       addressesToAdd: string[];
+ *       addressesToRemove: string[];
+ *     };
+ *   };
+ * }} config
+ *   - Configuration object containing the committee details and governance options.
+ *
+ * @returns {Promise<void>} A promise that resolves when the electorate has been
+ *   replaced.
+ */
+export const replaceAllElectorates = async (permittedPowers, config) => {
+  const { committeeName, voterAddresses, highPrioritySendersConfig } =
+    config.options;
+
+  const economicCommitteeCreatorFacet = await startNewEconomicCommittee(
+    permittedPowers,
+    {
+      options: {
+        committeeName,
+        committeeSize: values(voterAddresses).length,
+      },
+    },
+  );
+
+  const governedContractKitsMap =
+    await permittedPowers.consume.governedContractKits;
+  const psmKitMap = await permittedPowers.consume.psmKit;
+
+  const governanceDetails = [
+    ...[...governedContractKitsMap.values()].map(governedContractKit => ({
+      creatorFacet: governedContractKit.governorCreatorFacet,
+      label: governedContractKit.label,
+    })),
```

Comment:
@Chris-Hibbert this part notifies the price feeds of the new committee, right?

Since there's some doubt, I suppose it's worth testing. Maybe we're already doing that. Let's see...

Reply:
These lines add everything in `permittedPowers.consume.governedContractKits` to `governedContractKitsMap`. Everything in that list gets a new invitation.  I **thought** all the instances were also added to the charter, but I don't see that now. 

I don't think priceFeeds or scaledPriceAuthorities are in `permittedPowers.consume.governedContractKits`, so they have to be added separately. in #10074, a list of priceFeeds is passed in, and the list of scaledPriceAuthorities is created by looking for kits in `consume.contractKits` with the right installation.

Reply:
> I thought all the instances were also added to the charter, but I don't see that now.

If I am not wrong, isn't that covered in #10166 in `addGovernorsToEconCharter` definition?

Reply:
What I found in `governedContractKits`:
```bash 
[
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "ATOM-USD_price_feed",
      "publicFacet": {}
    },
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "provisionPool",
      "publicFacet": {}
    },
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "auctioneer",
      "publicFacet": {}
    },
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "AssetReserve",
      "publicFacet": {}
    },
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "VaultFactory",
      "privateArgs": {
        "feeMintAccess": {},
        "initialPoserInvitation": {},
        "initialShortfallInvitation": {},
        "marshaller": {},
        "storageNode": {}
      },
      "publicFacet": {}
    },
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "stATOM-USD_price_feed",
      "publicFacet": {}
    },
    {
      "adminFacet": {},
      "creatorFacet": {},
      "governor": {},
      "governorAdminFacet": {},
      "governorCreatorFacet": {},
      "instance": {},
      "label": "Crabble",
      "publicFacet": {}
    }
  ]
  

```

Reply:
It seemed to have ATOM and stATOM price feed

Reply:
> What I found in governedContractKits:

On which platform? Can we tell whether that's also true (for all priceFeeds) on mainNet?

Reply:
> On which platform? 
A3P
> Can we tell whether that's also true (for all priceFeeds) on mainNet?
I'm not sure at the moment, but I can look into it.